state/core_access.go

```diff
@@ -367,8 +355,21 @@ func (ca *CoreAccessor) BalanceForAddress(ctx context.Context, addr Address) (*B
 		return nil, fmt.Errorf("cannot convert %s into sdktypes.Int", string(value))
 	}
 	// verify balance
+	var proofOps *crypto.ProofOps
+	if result.GetProofOps() != nil {
```

Comment:
So, is there a case where the server gives us no proof? Shouldn't we disallow such a case?